velox/core/PlanNode.h

```diff
@@ -532,6 +532,27 @@ class AggregationNode : public PlanNode {
       bool ignoreNullKeys,
       PlanNodePtr source);
 
+  /// @param globalGroupingSets If the AggregationNode is over a GroupId node
+  /// then its possible that the grouping sets include global grouping sets
+  /// which have all null keys except for the groupId. These are for the global
+  /// aggregate values.
+  /// @param groupId If the AggregationNode is over a GroupId node,
+  /// then this is the groupId field.
+  /// GlobalGroupingSets and groupId trigger special handling when the data has
```

Comment:
The presence of globalGroupingSets triggers special logic when input dataset is empty (no rows). In that case, aggregation generates a single row with default global aggregate values per global grouping set.